### PR TITLE
Add form for PAC repository

### PR DESCRIPTION
**Pull request by rottencandy**
**Story**: 
https://issues.redhat.com/browse/ODC-6581

**Description**: 
- Add form for PAC Repository
- Create secret and link it when token is provided

**Screen shots / Gifs for design review**: 

**Unit test coverage report**: 

**Browser conformance**: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { TextInputTypes } from '@patternfly/react-core';
+import { FormikProps } from 'formik';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { FormFooter, FlexForm, FormBody, FormHeader, InputField } from '@console/shared';
+
+import './RepositoryForm.scss';
+
+type RepositoryFormProps = FormikProps<any>;
```
Comment (divyanshiGupta) [inline frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx:11]:
```suggestion
type RepositoryFormProps = FormikProps<FormikValues>;
```

**Comment by divyanshiGupta** [comment]
@saud i18n files needs to be updated. Please run `yarn i18n` to update these files.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx (hunk 1 above)
Comment (rottencandy) [reply frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx]: Updated.

**Comment by rottencandy** [comment]
According to the discussion, I've updated to use a simple form for creating the Repository CR.

File: frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            type={TextInputTypes.text}
+          />
+        </FormSection>
+      </FormBody>
+      <FormFooter
+        handleReset={handleReset}
+        errorMessage={status?.submitError}
+        isSubmitting={isSubmitting}
+        submitLabel={t('pipelines-plugin~Create')}
+        disableSubmit={!dirty || !_.isEmpty(errors)}
```
Comment (divyanshiGupta) [inline frontend/packages/pipelines-plugin/src/components/repository/RepositoryForm.tsx:50]:
```suggestion
        disableSubmit={!dirty || !_.isEmpty(errors) || isSubmitting}
```

File: frontend/packages/pipelines-plugin/src/components/repository/RepositoryFormPage.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <>
+      <Formik
+        initialValues={defaultRepositoryFormValues}
+        onSubmit={handleSubmit}
+        onReset={history.goBack}
+        validationSchema={repositoryValidationSchema(t)}
+      >
+        {(formikProps) => <RepositoryForm {...formikProps} />}
+      </Formik>
+    </>
```
Comment (divyanshiGupta) [inline frontend/packages/pipelines-plugin/src/components/repository/RepositoryFormPage.tsx:51]:
```suggestion
      <Formik
        initialValues={defaultRepositoryFormValues}
        onSubmit={handleSubmit}
        onReset={history.goBack}
        validationSchema={repositoryValidationSchema(t)}
      >
        {(formikProps) => <RepositoryForm {...formikProps} />}
      </Formik>
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Comment by divyanshiGupta** [comment]
@rottencandy I see `Create Repository` on the CRD page under instances tab. Shouldnt we show the form here as well and anywhere else this action is available to keep it consistent?

https://user-images.githubusercontent.com/20724543/162006601-896203f4-20e8-4243-9f09-08e2542b9b2c.mov

**Comment by rottencandy** [comment]
> @rottencandy I see `Create Repository` on the CRD page under instances tab. Shouldnt we show the form here as well and anywhere else this action is available to keep it consistent?

Was happening because the instances page is not namespaced so the form url was not being used. Added fallback to `default` namespace.

**Comment by divyanshiGupta** [comment]
Tested the PR locally and on a cluster. Works as expected.
/lgtm

**Review by divyanshiGupta — APPROVED** [review]

File: frontend/packages/pipelines-plugin/src/components/repository/repository-form-utils.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    Math.floor((1 + Math.random()) * 0x10000)
+      .toString(16)
+      .substring(1);
+
+  const data: SecretKind = {
+    apiVersion: SecretModel.apiVersion,
+    kind: SecretModel.kind,
+    metadata: {
+      name: `${repositoryName}-token-${s4()}`,
+    },
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/repository/repository-form-utils.ts:42]: Small improvement idea: Kubernetes supports a special metadata `generateName` attributes to create random suffixed resources. You can simplify this like this:

```suggestion
  const data: SecretKind = {
    apiVersion: SecretModel.apiVersion,
    kind: SecretModel.kind,
    metadata: {
      generateName: `${repositoryName}-token-`,
    },
```

Wdyt?

**Comment by jerolimov** [comment]
@karthikjeeyar Everything else is fine for me. Can you approve it if it's also fine for you?

/uncc

File: frontend/packages/pipelines-plugin/src/components/repository/repository-form-utils.ts (hunk 4 above)
Comment (rottencandy) [reply frontend/packages/pipelines-plugin/src/components/repository/repository-form-utils.ts]: Thanks, did not know about this.

**Review by jerolimov — APPROVED** [review]
Direction is fine for me. Didn't add lgtm because I don't tested the details. :)

**Comment by jerolimov** [comment]
Tested now locally, works fine. 

/lgtm

**Pull request merged**